### ISSUE TITLE
When you set microbatch, it will not leak privacy?

### ISSUE THREAD

**Issue by Suetsumu** [dayu11/Differentially-Private-Deep-Learning#7]
https://github.com/dayu11/Differentially-Private-Deep-Learning/blob/73555c8eb96eabfb441172525bd32c8d0a771d37/vision/RGP/cifar_train.py#L194
You clip gradient and accumulate it into p.grad, but when you operate loss.backward(), average gradient will also be put into p.grad automatically, so I think you mix average gradient without any DP operation and clipped sample gradient.
